client/src/components/SignUpForm/SignUpForm.tsx

```diff
@@ -0,0 +1,153 @@
+import { Field, Form, Formik, FormikState } from 'formik';
+import * as Yup from 'yup';
+import React, { useState } from 'react';
+import styles from './SignUpForm.module.scss';
+import { SignUpDto } from '../../dtos/auth/SignUp.dto';
+import { CustomButton } from '../common/CustomButton/CustomButton';
+import { CustomInput } from '../common/CustomInput/CustomInput';
+import { useAppDispatch } from '../../hooks/redux';
+import { signUp } from '../../stateManager/actionCreators/auth';
+import { Preloader } from '../common/Preloader/Preloader';
+import { phoneValidateRegExp } from '../../utils/validation/regularExpressions';
+import { phoneNumberMask } from '../../utils/masks/phoneNumberMask';
+
+interface IProps {}
+
+export const SignUpForm: React.FC<IProps> = ({}) => {
+    const dispatch = useAppDispatch();
+    const [formSendError, setFormSendError] = useState('');
+
+    const formInitialValues: SignUpDto = {
+        email: '',
+        password: '',
+        firstName: '',
+        lastName: '',
+        phone: '',
+    };
+
+    const submit = async (
```

Comment:
Note: callbacks is better to be named same as parameter so it is better to name it like `onFormSubmit`